stories/pagination/pagination.stories.js

```diff
@@ -18,30 +18,60 @@ Element | Modifier/class | Description
+Next page button | \`fd-pagination__button\` | The next arrow that users can use to navigate forward. This arrow is disabled when on the last page.
+Last page button | \`fd-pagination__button--mobile\` | The last page button that users can use to navigate to the last page. This button is disabled when on the last page. Shown only on mobile.
+Per page | \`fd-pagination__per-page\` | The area where items per page select & its label are placed
+Per page label | \`fd-pagination__per-page-label\` | Per page section label. Hidden on mobile.
 <br>
 <br>
 
       `,
         tags: ['a11y', 'theme'],
-        components: ['pagination']
+        components: ['select', 'button', 'icon', 'form-label', 'input', 'pagination', 'popover', 'list']
     }
 };
 
-export const firstPage = () => `<div class="fd-pagination">
-  <span class="fd-pagination__total">30 items</span>
-  <nav class="fd-pagination__nav">
-      <a href="#" class="fd-pagination__link fd-pagination__link--previous" aria-label="Previous"aria-disabled="true"></a>
-      <a href="#" class="fd-pagination__link is-selected" >1</a>
-      <a href="#" class="fd-pagination__link">2</a>
-      <a href="#" class="fd-pagination__link">3</a>
-      <a href="#" class="fd-pagination__link fd-pagination__link--next" aria-label="Next"
-      aria-disabled="false"></a>
-  </nav>
+export const firstPage = () => `<style>*[dir='rtl'] .fd-pagination__button { transform: rotate(180deg) !important; }</style>
+<div class='fd-pagination'>
+    <nav class='fd-pagination__nav'>
+        <div style=''>
+        
+        </div>
+        <a href='#' class='fd-button fd-button--compact fd-button--transparent fd-pagination__button fd-pagination__button--mobile' aria-label='First' aria-disabled='true'>
+            <i class='sap-icon sap-icon--media-rewind'></i>
+        </a>
+        
+        <a href='#' class='fd-button fd-button--compact fd-button--transparent fd-pagination__button' aria-label='Previous' aria-disabled='true'>
+            <i class='sap-icon sap-icon--navigation-left-arrow'></i>
+        </a>
+        
+        <label class='fd-form-label fd-pagination__input-label' for='firstPageInput'>Page:</label>
+        
+        <input id='firstPageInput' class='fd-input fd-input--compact fd-pagination__input' type='number' min='1' max='3' value='1' required />
+        
+        <a href='#' class='fd-button fd-button--compact fd-button--transparent fd-pagination__link'>2</a>
+        
+        <a href='#' class='fd-button fd-button--compact fd-button--transparent fd-pagination__link'>3</a>
+        
+        <a href='#' class='fd-button fd-button--compact fd-button--transparent fd-pagination__button' aria-label='Next' aria-disabled='false'>
+            <i class='sap-icon sap-icon--navigation-right-arrow'></i>
+        </a>
+        
+        <a href='#' class='fd-button fd-button--compact fd-button--transparent fd-pagination__button fd-pagination__button--mobile' aria-label='Last' aria-disabled='false'>
```

Comment:
`fd-pagination__button--mobile`?

Reply:
Yea, it's for buttons that are shown only on a mobile.

Reply:
My question is why are you combining it with `fd-button--compact`? `--compact` and `--mobile` modifiers usually define different metrics for a component. For example, in compact (default) mode, the height of an element is 24px, but in mobile mode it changes to 32px. So it makes no sense to have a modifier class on an element to set it to compact mode and then to mobile.

Reply:
Fixed.